crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -789,16 +808,30 @@ impl SlidingSync {
                     }
                     unsubs
                 };
+                let timeout = Duration::from_secs(30);
+
+                // implement stickiness by only sending extensions if they have changed since the last time we sent them
+                // TODO: use PartialEq instead rather than comparing JSON forms of ExtensionsConfig?
```

Comment:
Once https://github.com/ruma/ruma/pull/1429 lands this can be updated to

```patch
--- a/crates/matrix-sdk/src/sliding_sync.rs
+++ b/crates/matrix-sdk/src/sliding_sync.rs
@@ -811,10 +811,7 @@ impl SlidingSync {
                 let timeout = Duration::from_secs(30);
 
                 // implement stickiness by only sending extensions if they have changed since the last time we sent them
-                // TODO: use PartialEq instead rather than comparing JSON forms of ExtensionsConfig?
-                let extensions_json = serde_json::to_string(&self.extensions.lock().unwrap().clone()).unwrap();
-                let sent_extensions_json = serde_json::to_string(&self.sent_extensions.lock().unwrap().clone()).unwrap();
-                let extensions = if extensions_json == sent_extensions_json {
+                let extensions = if self.extensions.lock().unwrap().clone() == self.sent_extensions.lock().unwrap().clone() {
                     None
                 } else {
                     self.extensions.lock().unwrap().clone()
```

Reply:
If you bump the git rev for Ruma in the toplevel `Cargo.toml`, you can apply this.